### PR TITLE
create card api service updated

### PR DESCRIPTION
**Pull request by Caleb-Nieto**
### Feature / Bug Description
create card api service updated

### Solution
create card api service updated

### Notes
The responsible id is requested because is not null in the database.
For now there are only 6 types of evidence validation in the switch cause table Card see the image 1

### Tickets
[WMA-105](https://cdentalcaregroup.atlassian.net/browse/WMA-105)

### Screenshots / Screencasts
image 1
image
Captura de pantalla 2024-06-21 170313
Captura de pantalla 2024-06-21 170328

[WMA-105]: https://cdentalcaregroup.atlassian.net/browse/WMA-105?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/modules/card/card.service.ts [hunk 1]
```diff
@@ -119,87 +126,73 @@ export class CardService {
...
+      const { siteCardId } = lastInsertedCard[0];
+
+      const card = await this.cardRepository.create({
+        ...createCardDTO,
+        siteCardId: siteCardId + 1,
+        siteCode: site.siteCode,
+        cardTypeColor: cardType.color,
+        areaName: area.name,
+        level: area.level,
+        superiorId: Number(area.superiorId) === 0 ? area.id : area.superiorId,
```
Comment (imadiaz) [inline src/modules/card/card.service.ts:142]: we're going to send you the superiorId in the CLIENT side

File: src/modules/card/card.service.ts [hunk 2]
```diff
@@ -119,87 +126,73 @@ export class CardService {
...
+        priorityCode: priority.priorityCode,
+        priorityDescription: priority.priorityDescription,
+        cardTypeMethodology: cardType.cardTypeMethodology,
+        cardTypeMethodologyName: cardType.methodology,
+        cardTypeName: cardType.name,
+        preclassifierCode: preclassifier.preclassifierCode,
+        preclassifierDescription: preclassifier.preclassifierDescription,
+        creatorName: creator.name,
+        responsableName: responsible.name,
```
Comment (imadiaz) [inline src/modules/card/card.service.ts:151]: can se set this nullable? or is required?

File: src/modules/card/card.service.ts [hunk 3]
```diff
@@ -119,87 +126,73 @@ export class CardService {
...
+      });
 
-     return await this.cardRepository.save(createCardDTO)
+      await this.cardRepository.save(card);
+      lastInsertedCard = await this.cardRepository.find({order: {id: 'DESC'}, take: 1})
+      const cardAssignEvidences = lastInsertedCard[0]
+
+      await Promise.all(createCardDTO.evidences.map(async (evidence) => {
+        switch (evidence.type) {
+          case 'AUCR':
```
Comment (imadiaz) [inline src/modules/card/card.service.ts:163]: can u use constants here?

File: src/modules/card/card.service.ts (hunk 2 above)
Comment (imadiaz) [reply src/modules/card/card.service.ts:151]: looks like responsable and responsable id we update it in another service, can se remove it? set null or EMPTY values

Comment (Caleb-Nieto) [reply src/modules/card/card.service.ts:151]: Im not seeing the database right now but at least responsible id is not null

File: src/modules/card/card.service.ts (hunk 3 above)
Comment (Caleb-Nieto) [reply src/modules/card/card.service.ts:163]: got it

File: src/modules/card/card.service.ts (hunk 2 above)
Comment (imadiaz) [reply src/modules/card/card.service.ts:151]: Let me see if i can make it nullable

Comment (imadiaz) [reply src/modules/card/card.service.ts:151]: I make them nullables, can u check it again?

Comment (Caleb-Nieto) [reply src/modules/card/card.service.ts:151]: also de name is not null

Comment (imadiaz) [reply src/modules/card/card.service.ts:151]: can u check it again the values?

**Review by imadiaz — APPROVED** [review]
lgtm

**Pull request merged**